test/unit/test_arguments.py

```diff
@@ -321,29 +325,40 @@ def grouper(iterable, n, fillvalue=None):
         return zip_longest(fillvalue=fillvalue, *args)
 
     params = params.split()
-    regular_files_params = {'-vs', '-vt', '-t', '-s', '--source', '--target', '--validation-source', '--validation-target'}
+    regular_files_params = {'-vs', '-vt', '-t', '-s', '--source', '--target',
+                            '--validation-source', '--validation-target'}
+    folder_params = {'--prepared-data', '-d'}
     to_unlink = set()
     for arg, val in grouper(params, 2):
         if arg in regular_files_params and not os.path.isfile(val):
-            to_unlink.add((val, open(val, 'w')))
+            open(val, 'w').close()
+            to_unlink.add(val)
+        if arg in folder_params:
+            os.mkdir(val)
+            to_unlink.add(val)
     return to_unlink
 
 
-def _delete_argument_values_that_must_be_files(to_unlink):
+def _delete_argument_values_that_must_be_files_or_dirs(to_unlink):
     """
-    Close and delete previously created files.
+    Close and delete previously created files or directories.
     """
-    for name, f in to_unlink:
-        f.close()
-        os.unlink(name)
+    for name in to_unlink:
+        if os.path.isfile(name):
+            os.unlink(name)
+        else:
+            os.rmdir(name)
 
 
 def _test_args(test_params, expected_params, args_func):
     test_parser = argparse.ArgumentParser()
     args_func(test_parser)
-    created = _create_argument_values_that_must_be_files(test_params)
-    parsed_params = test_parser.parse_args(test_params.split())
-    _delete_argument_values_that_must_be_files(created)
+    # TODO: create temporary directory?
```

Comment:
lets use a 'with TemporaryDicrectory'...

Reply:
actually the issue is that if we did we would either need to change the working directory to the temporary working dir or change the arguments to change `--source test_src` to `--source q34owi5n/test_src`.

Reply:
so I'd just leave it the way it is right now.